analyzers/tests/SonarAnalyzer.Test/TestCases/CollectionQuerySimplification.CSharp9.cs

```diff
@@ -2,10 +2,87 @@
 using System.Collections.Generic;
 using System.Collections;
 using System.Linq;
+using Microsoft.EntityFrameworkCore;
 
 List<object> list = null;
 
 list.Select(static (element, col) => element is Int32 and > 21).Any(element => element != null); // Compliant
 list.Select((object _, int _) => 1).Any(element => element != null); // Compliant
 
 list.Select((element, col) => element as object).Any(element => element != null);  //Noncompliant {{Use 'OfType<object>()' here instead.}}
+
+
+// https://github.com/SonarSource/sonar-dotnet/issues/3604
+public class EntityFrameworkReproGH3604
+{
+    public class MyEntity
+    {
+        public int Id { get; set; }
+    }
+
+    public class FirstContext : DbContext
+    {
+        public DbSet<MyEntity> MyEntities { get; set; }
+    }
+
+    public void GetEntities(FirstContext dbContext)
```

Comment:
Fully qualify the names, so we can see at one glance what namespace they belong to. This is especially important for DbSet as it comes from two namespace, right?

Reply:
For `dbContext` it is a local type.
I made sure everything related to EF (Core or not) are fully qualified.